### PR TITLE
[JENKINS-67028] Fix missing project hyperlink in build history

### PR DESCRIPTION
**Pull request by NotMyFault**
Fixes https://issues.jenkins.io/browse/JENKINS-67028

### Proposed changelog entries

* Fix missing hyperlink in build history (regression in 2.314).

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [X] (If applicable) Jira issue is well described
- [X] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade

### Desired reviewers

@MarkEWaite 

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [x] There are at least 2 approvals for the pull request and no outstanding requests for change
- [x] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [x] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [x] Proper changelog labels are set so that the changelog can be generated automatically
- [x] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [x] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

**Comment by MarkEWaite** [comment]
Thanks very much @NotMyFault !  I've tested interactively and confirmed that the hyperlink is inserted in the build history and links as expected to the build details page.

**Review by MarkEWaite — APPROVED** [review]

**Review by oleg-nenashev — APPROVED** [review]
Looks good to me taking the neighboring changes. Thanks for the patch!

We may merge it in 24 hours if there is no negative feedback. Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process

**Comment by MarkEWaite** [comment]
> We may merge it in 24 hours if there is no negative feedback. Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process

Let's wait to merge this until after the Jenkins 2.319 weekly release is delivered on Thursday, Nov 4, 2021.  Merging earlier won't make the fix available any sooner and delaying the merge will allow each PR submitter to handle the merge rather than placing the merge burden on the Jenkins security team.

**Pull request merged**
